app/controllers/saml_completion_controller.rb

```diff
@@ -0,0 +1,27 @@
+# Handles the INTERNAL redirect which happens when a service provider authentication request
+# is passed through the IDP's authentication flow (sign-in, MFA, etc.)
+# The original request was saved to the sp_session object, so we retrieve it to pass on the
+# original request url in the form of a POST request to the SamlIdpController#auth method
+class SamlCompletionController < ApplicationController
+  # Pass the original service provider request to the main SamlIdpController#auth method
+  # via a POST with form parameters replacing the url query parameters
+  def index
+    if sp_session.present?
+      request_url = URI(sp_session[:request_url])
+      path_year = request_url.path[-4..-1]
+      path_method = "api_saml_finalauthpost#{path_year}_url"
```

Comment:
maybe as like a `FIXME`/future facing note, we could come back and try to redo `routes.rb` so that the year is a path param, so we could jstu do `params[:path_year]` instead of having to extract the year frmo the URL like this?

I just confirmed that this works

```
  get "/api/saml/auth(:path_year)" => 'saml_idp#auth', constraints: { path_year: SamlEndpoint.suffixes }
```

Reply:
I like the idea of fixing all of them at once, so I want to log a JIRA for that work. Great suggestion! https://cm-jira.usa.gov/browse/LG-7554